Sources/StreamChat/APIClient/Endpoints/Payloads/IdentifiableModel.swift

```diff
@@ -0,0 +1,61 @@
+//
+// Copyright © 2022 Stream.io Inc. All rights reserved.
+//
+
+import CoreData
+import Foundation
+
+typealias PreWarmedCache = [String: [String: NSManagedObjectID]]
```

Comment:
This comment will apply throughout this file

Can we make these more readable by typealiasing a bunch of stuff
```swift
typealias DatabaseName = String
typealias DatabaseId = String
```
then
```
[String: Set<String>] => [DatabaseName: Set<DatabaseId>]
[String: [String: NSManagedObjectId]] => [DatabaseName: [DatabaseId: NSManagedObjectId]]
```
also
```
protocol IdentifiableModel {
     static var className: DatabaseName { get }
     static var idKeyPath: String? { get }
     static func id(for model: NSManagedObject) -> DatabaseId?
 }
```

Reply:
You can do that, but then you need to get the results in the same order to be able to match id to objectID. Which doesn't seem to be the case.

Reply:
Can you clarify? Why do we need to get the results in the same order? It's just eye-sugar

Reply:
I swear I put this comment in another thread..